tensorflow_io/genome/python/ops/genome_ops.py

```diff
@@ -17,6 +17,72 @@
 from __future__ import division
 from __future__ import print_function
 
+import tensorflow as tf
 from tensorflow_io.core.python.ops import core_ops
 
 read_fastq = core_ops.read_fastq
+
+
+
+@tf.function
+def _nucleotide_to_onehot(nucleotide):
+  """Encodes a nucleotide using a one hot encoding."""
+  a_onehot = tf.constant([1, 0, 0, 0])
+  c_onehot = tf.constant([0, 1, 0, 0])
+  g_onehot = tf.constant([0, 0, 1, 0])
+  t_onehot = tf.constant([0, 0, 0, 1])
+  error_onehot = tf.constant([1, 1, 1, 1])
+
+  if tf.math.equal(nucleotide, tf.constant(b'A')):  # pylint: disable=no-else-return
+    return a_onehot
+  elif tf.math.equal(nucleotide, tf.constant(b'C')):
+    return c_onehot
+  elif tf.math.equal(nucleotide, tf.constant(b'G')):
+    return g_onehot
+  elif tf.math.equal(nucleotide, tf.constant(b'T')):
+    return t_onehot
+  else:
+    # TODO(suyashkumar): how best to raise error from within tf.function?
+    return error_onehot
+
+
+@tf.function
+def sequences_to_onehot(sequences):
+  """Convert DNA sequences into a one hot nucleotide encoding.
+
+  Each nucleotide in each sequence is mapped as follows:
+  A -> [1, 0, 0, 0]
+  C -> [0, 1, 0, 0]
+  G -> [0 ,0 ,1, 0]
+  T -> [0, 0, 0, 1]
+
+  If for some reason a non (A, T, C, G) character exists in the string, it is
+  currently mapped to a error one hot encoding [1, 1, 1, 1].
+
+  Args:
+    sequences: A tf.string tensor where each string represents a DNA sequence
+
+  Returns:
+    tf.RaggedTensor: The output sequences with nucleotides one hot encoded.
+  """
+  all_onehot_nucleotides = tf.TensorArray(
+      dtype=tf.int32, size=0, dynamic_size=True)
```

Comment:
todo: see if I can statically calculate this size in a way that tf.function is okay with